tensorflow/python/framework/python_op_gen_test.cc

```diff
@@ -23,17 +23,437 @@ limitations under the License.
 namespace tensorflow {
 namespace {
 
-TEST(PythonOpGen, Basic) {
+void ExpectHasSubstr(const string& s, const string& expected) {
+  EXPECT_TRUE(absl::StrContains(s, expected))
+      << "'Generated ops does not contain '" << expected << "'";
+}
+
+void ExpectDoesNotHaveSubstr(const string& s, const string& expected) {
+  EXPECT_FALSE(absl::StrContains(s, expected))
+      << "'Generated ops contains '" << expected << "'";
+}
+
+void ExpectSubstrOrder(const string& s, const string& before,
+                       const string& after) {
+  int before_pos = s.find(before);
+  int after_pos = s.find(after);
+  ASSERT_NE(std::string::npos, before_pos);
+  ASSERT_NE(std::string::npos, after_pos);
+  EXPECT_LT(before_pos, after_pos)
+      << before << "' is not before '" << after;
+}
+
+TEST(PythonOpGen, TypeAnnotateAllOps) {
   OpList ops;
   OpRegistry::Global()->Export(false, &ops);
 
   ApiDefMap api_def_map(ops);
 
-  string code = GetPythonOps(ops, api_def_map, {}, "");
+  std::unordered_set<string> type_annotate_ops;
+  for (const auto& op : ops.op()) {
+    type_annotate_ops.insert(op.name());
+  }
+
+  string code = GetPythonOps(ops, api_def_map, {}, "", type_annotate_ops);
+
+  const string all_types = ", _dtypes.BFloat16, _dtypes.Bool, _dtypes.Complex128, _dtypes.Complex64, "
```

Comment:
Be careful about making these checks too exhaustive. We don't want modifications to the code to require updating this test too often.